### ISSUE TITLE
A false negative about the rule BC_IMPOSSIBLE_DOWNCAST_OF_TOARRAY

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1759]
**SpotBugs Version:** 4.4.2
**Rule:** [BC_IMPOSSIBLE_DOWNCAST_OF_TOARRAY](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#bc-impossible-downcast-of-toarray-result-bc-impossible-downcast-of-toarray)

Hi! I find a false negative about this rule. SpotBugs should have reported a warning at line 4. This program can lead to a running exception.

**Code Sample**
```java
ArrayList<Integer> lst = new ArrayList<Integer>();
Integer[] res;
public Integer[] asArray() {
    res = (Integer[]) ((Object[])(lst.toArray()));  // should report a warning here
    return this.res;
}
```
**Expected outcome:**
SpotBugs should report a violation at line 4, but doesn't. This is a false-negative.

**Running SpotBugs through:** [CLI]
